modules/rostests/apitests/ntdll/NtQueryInformationProcess.c

```diff
@@ -186,6 +186,348 @@ Test_ProcessTimes(void)
+                                       NULL,
+                                       sizeof(QUOTA_LIMITS_EX) - 1,
+                                       NULL);
+    ok_hex(Status, STATUS_INFO_LENGTH_MISMATCH);
+
+    /* Right buffer size but NULL pointer */
+    Status = NtQueryInformationProcess(NtCurrentProcess(),
+                                       ProcessQuotaLimits,
+                                       NULL,
+                                       sizeof(QUOTA_LIMITS_EX),
+                                       NULL);
+    ok_hex(Status, STATUS_ACCESS_VIOLATION);
+
+    /* Buffer too large */
+    Status = NtQueryInformationProcess(NtCurrentProcess(),
+                                       ProcessQuotaLimits,
+                                       NULL,
+                                       sizeof(QUOTA_LIMITS_EX) + 1,
+                                       NULL);
+    ok_hex(Status, STATUS_INFO_LENGTH_MISMATCH);
+
+    /* Buffer too small, ask for length */
+    Length = 0x55555555;
+    Status = NtQueryInformationProcess(NtCurrentProcess(),
+                                       ProcessQuotaLimits,
+                                       NULL,
+                                       sizeof(QUOTA_LIMITS_EX) - 1,
+                                       &Length);
+    ok_hex(Status, STATUS_INFO_LENGTH_MISMATCH);
+    ok_dec(Length, 0x55555555);
+
+    /* Valid parameters, no return length */
+    RtlFillMemory(&QuotaLimitsEx1, sizeof(QuotaLimitsEx1), 0x55);
+    Status = NtQueryInformationProcess(NtCurrentProcess(),
+                                       ProcessQuotaLimits,
+                                       &QuotaLimitsEx1,
+                                       sizeof(QUOTA_LIMITS_EX),
+                                       NULL);
+    ok_hex(Status, STATUS_SUCCESS);
+
+    /* Trace the returned data (1) */
+    trace("QuotaLimitsEx1.PagedPoolLimit = %Iu\n", QuotaLimitsEx1.PagedPoolLimit);
+    trace("QuotaLimitsEx1.NonPagedPoolLimit = %Iu\n", QuotaLimitsEx1.NonPagedPoolLimit);
+    trace("QuotaLimitsEx1.MinimumWorkingSetSize = %Iu\n", QuotaLimitsEx1.MinimumWorkingSetSize);
+    trace("QuotaLimitsEx1.MaximumWorkingSetSize = %Iu\n", QuotaLimitsEx1.MaximumWorkingSetSize);
+    trace("QuotaLimitsEx1.PagefileLimit = %Iu\n", QuotaLimitsEx1.PagefileLimit);
+    trace("QuotaLimitsEx1.TimeLimit = %I64u\n", QuotaLimitsEx1.TimeLimit.QuadPart);
+    //trace("QuotaLimitsEx1.WorkingSetLimit = %Iu\n", QuotaLimitsEx1.WorkingSetLimit); // Not used on Win2k3
+    trace("QuotaLimitsEx1.Flags = %lx\n", QuotaLimitsEx1.Flags);
+    trace("QuotaLimitsEx1.CpuRateLimit.RateData = %lx\n", QuotaLimitsEx1.CpuRateLimit.RateData);
+
+    /* Again, this time with a return length */
+    Length = 0x55555555;
+    RtlFillMemory(&QuotaLimitsEx2, sizeof(QuotaLimitsEx2), 0x55);
+    Status = NtQueryInformationProcess(NtCurrentProcess(),
+                                       ProcessQuotaLimits,
+                                       &QuotaLimitsEx2,
```

Comment:
Why using a separate variable, instead of re-using the QuotaLimitsEx1 one (after resetting its contents)?